scripts/check_integrity.py

```diff
@@ -0,0 +1,251 @@
+            if 'journal' in entry:
+                venue = entry['journal']
+                if isinstance(venue, str):
+                    venuePart = venue.upper()
+                else:
+                    venuePart = venue.expr[0].name.upper()
+        if 'year' in entry:
+            year = entry['year']
+            if isinstance(year, str):
+                yearPart = year[2:4]
+            else:
+                yearPart = year.expr[0].name[2:4]
+        key = namePart + venuePart + yearPart
+        if (not org_key.casefold().startswith(key.casefold())):
+            add_problem(entry, 'incorrect_bib_key', org_key + " != " + key)
+       
+
+if __name__ == '__main__':
+    with open(get_path("check_integrity_config.yaml"), "r") as stream:
+        try:
+            settings = yaml.safe_load(stream)
+            #print(settings)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    with open(get_path("required_fields.yaml"), "r") as stream:
+        try:
+            fields = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    with open(get_path("string_fields.yaml"), "r") as stream:
+        try:
+            string_fields = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    # Check capitilization
+    # Check double braces for title
+    # Check string fields
+    # Check author first name, second name order
+    # Check bib keys
+
+    # Check entry order: year, type, venue, month, key
+    # Check notes up to date
+
+    parser = BibTexParser(common_strings=False,interpolate_strings=False)
+    parser.homogenise_fields=False
+    parser.ignore_nonstandard_types=False
+
+    entrytypes = fields.keys()
+    entries = {}
+    problems = {}
+
+    with open(database_file) as bibtex_file:
+        bib_database = bibtexparser.load(bibtex_file, parser)
+        #df = pd.DataFrame(bib_database.entries)
+        #df.to_csv('literature.csv', index=False)
+
+        for entry in bib_database.entries:
+            check_string_fields(entry)
+            check_field_types_exists(entry)
+            check_title(entry)
+            check_author(entry)
+            check_key(entry)
+            check_pages(entry)
+
+        new_strings = {}
+        for entry in bib_database.entries:
+            gen_string_fields(entry)
+
+    count = 0
+    for entry_problems in problems.items():
+        count = count + 1
+        print(str(count) + ": " + entry_problems[0])
+        for entry_problem in entry_problems[1]:
+            print('\t' + str(entry_problem))
+
+    if is_set('print_missing_strings'):
+        ns = list(new_strings.values())
+        ns.sort(key=lambda a: a[0])
+        field_type = None
+        for new_string in ns:
+            if field_type != new_string[0]:
+                field_type = new_string[0]
+                print(field_type)
+            print("@String{" + new_string[1] + " = \""+ new_string[2] +"\"}")
+
```

Comment:
It would be great to have some documentation about what the script does as a whole? What do the yaml-files mean exactly and how are their options processed. What does the script produce? Does it mutate any files? Does it generate any files? Does it (just) report errors to terminal?